### DISCUSSION TITLE
CSI Storage Driver migration and upgrade impact

### DISCUSSION THREAD

**Discussion by tomasr** [Azure/AKS#2504] [Q&A]
With the (https://azure.microsoft.com/en-us/updates/general-availability-csi-storage-driver-support-on-azure-kubernetes-service/)[GA of the CSI Storage Driver] and it coming up as default in Kubernetes 1.21, I'm wondering if we could have some guidance as to what this impacts?

Our scenario right now is we rely on both the "azurefile" and "azurefile-premium" storage classes, and have a large number of persistent volumes and persistent volume claims on them. So I'm a bit concerned as to what is going to break when trying to upgrade.

Am I wrong in assuming I'm likely going to have to delete all PVCs and PVs, possibly create new storage classes on CSI, and then recreate everything? If so, any particular things to worry about?

**Comment by tomasr** [comment]
Sort of answering my own question here after doing an initial test.

When upgrading to 1.21, you still have the old storage classes, but now you also have the new CSI-based ones:

![image](https://user-images.githubusercontent.com/16179/133324385-1b8f532c-3930-4f66-bb23-b07acb8bf45a.png)

Existing persistent volumes on the old storage classes appear to be left unaffected, so that much is good. I would assume next step is to actually manually migrate all your PVs/PVCs over to the CSI storage classes, but I don't have enough information to even attempt that yet on my side.

**Reply by ankurjains** [reply]
that a good insight( i was in doubt on same as in doc no where mention about this)  we are going to migrate soon to 1.21
